### PR TITLE
add 2 byte field for extended advertisement event kind

### PR DESCRIPTION
**Pull request by lulf**
The extended advertisement report has a 2 octet event kind field. One of the fields is actually a 2 bit value, which the macros don't support, hence the two fields. What's the preferred way to do that you think? I can see if I can expand the macro or maybe there is a more elegant way?

### CODE REVIEW

**Comment by lulf** [comment]
I've added the enum and methods for setting values using them.

File: src/param/le.rs [hunk 1]
```diff
@@ -496,29 +494,24 @@ pub enum LeExtAdvDataStatus {
...
-        match (self.data_status_1(), self.data_status_2()) {
-            (false, false) => LeExtAdvDataStatus::Complete,
-            (false, true) => LeExtAdvDataStatus::IncompleteMoreExpected,
-            (true, false) => LeExtAdvDataStatus::IncompleteTruncated,
+        let data_status = self.0[0] >> 5 & 0x03;
+        match data_status {
+            0 => LeExtAdvDataStatus::Complete,
+            1 => LeExtAdvDataStatus::IncompleteMoreExpected,
+            2 => LeExtAdvDataStatus::IncompleteTruncated,
             _ => unreachable!(),
```
Comment (alexmoon) [inline src/param/le.rs:502]: I don't think we should panic on unexpected data. A newer version of the spec might assign meaning to that value. I'd recommend adding a `Reserved` variant to the enum.

Comment (lulf) [reply src/param/le.rs]: Ok, done. Currently reserved sets 0xff which seems to be the value Zephyr uses for ' invalid status'.

Comment (lulf) [reply src/param/le.rs]: I mean 0b11 of course... :)

File: src/param/le.rs [hunk 2]
```diff
@@ -508,6 +509,7 @@ impl LeExtAdvEventKind {
             LeExtAdvDataStatus::Complete => 0,
             LeExtAdvDataStatus::IncompleteMoreExpected => 1,
             LeExtAdvDataStatus::IncompleteTruncated => 2,
+            LeExtAdvDataStatus::Reserved => 0xFF,
```
Comment (alexmoon) [inline src/param/le.rs]: This should be 3. Otherwise it will affect other bits in the bitfield.

Comment (lulf) [reply src/param/le.rs]: Please refresh :)

Comment (lulf) [reply src/param/le.rs]: (the fix was already pushed, as i realized i made the mistake right after i push initially)

**Review by alexmoon — APPROVED** [review]

**Pull request merged**